java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -401,6 +401,8 @@ public TransferPair makeTransferPair(ValueVector target) {
   @Override
   public void copyFrom(int inIndex, int outIndex, ValueVector from) {
     Preconditions.checkArgument(this.getMinorType() == from.getMinorType());
+    Preconditions.checkArgument(inIndex >= 0 && inIndex < from.getValueCount(),
+        "Invalid inIndex: %s", inIndex);
```

Comment:
I am not sure if it is a good idea to add check here. As this method can be performance critical, we decided to add minimum checks here, to compare minor types. 
Please see the discussion here: https://github.com/apache/arrow/pull/5019

Reply:
Thanks for clarify, I reverted changes in this methods.